### PR TITLE
lib/system: More Types of CPUs Added

### PR DESCRIPTION
**Pull request by vincrusher**
I have added more types of CPUs to try to support.

These could be useful for porting Nix packages to more platforms.

### CODE REVIEW

File: lib/systems/parse.nix [hunk 1]
```diff
@@ -80,6 +80,7 @@ rec {
     armv8r   = { bits = 32; significantByte = littleEndian; family = "arm"; version = "8"; };
     armv8m   = { bits = 32; significantByte = littleEndian; family = "arm"; version = "8"; };
     aarch64  = { bits = 64; significantByte = littleEndian; family = "arm"; version = "8"; };
+    aarch64_be = { bits = 64; significantByte = bigEndian; family = "arm" version = "8"; };
```
Comment (Mic92) [inline lib/systems/parse.nix]:
```suggestion
    aarch64_be = { bits = 64; significantByte = bigEndian; family = "arm"; version = "8"; };
```

File: lib/systems/parse.nix [hunk 2]
```diff
@@ -101,6 +103,8 @@ rec {
 
     wasm32   = { bits = 32; significantByte = littleEndian; family = "wasm"; };
     wasm64   = { bits = 64; significantByte = littleEndian; family = "wasm"; };
+    
+    alpha    = { bits = 64; significantBye = littleEndian; family = "alpha"; };
```
Comment (xeji) [inline lib/systems/parse.nix]: typo: `significantByte` to fix eval

File: lib/systems/examples.nix [hunk 3]
```diff
@@ -112,11 +112,26 @@ rec {
     config = "aarch64-none-elf";
     libc = "newlib";
   };
+  
+  aarch64be-embedded = {
```
Comment (Mic92) [inline lib/systems/examples.nix]: @Ericson2314 do we have any requirements for systems added in this file?

**Comment by matthewbauer** [comment]
Yes- the parse.nix changes are good but the changes to examples.nix might require some more discussion. Adding "alpha-embedded" definitely seems good because none of the other ones are alpha but I'm not so sure about the power/aarch64 "endian" cases. We don't need to cover every possible combination here - just the most common ones. I think both of those processors can switch between the two.

File: lib/systems/examples.nix (hunk 3 above)
Comment (Ericson2314) [reply lib/systems/examples.nix]: Not really! The only thing would be trying to keep the `config`s unique.

**Comment by vincrusher** [comment]
They do support big-endian and little-endian. I thought adding support for both as some people might like to use little over big or vice versa.

**Comment by matthewbauer** [comment]
Looks fine for now!

**Pull request merged**